File: password_checker.py
password = input("Enter your password: ")

length = len(password)
has_upper = False
has_lower = False
has_digit = False
has_special = False

special_characters = "!@#$%^&*"

for char in password:
    if char.isupper():
        has_upper = True
    elif char.islower():
        has_lower = True
    elif char.isdigit():
        has_digit = True
    elif char in special_characters:
        has_special = True

score = 0

if length >= 8:
    score = score + 1
if has_upper:
    score = score + 1
if has_lower:
    score = score + 1
if has_digit:
    score = score + 1
if has_special:
    score = score + 1

print("\nPassword Analysis")
print("Length:", length)
print("Has uppercase letter:", has_upper)
print("Has lowercase letter:", has_lower)
print("Has number:", has_digit)
print("Has special character:", has_special)

if score <= 2:
    print("\nPassword Strength: Weak")
elif score <= 4:
    print("\nPassword Strength: Medium")
else:
    print("\nPassword Strength: Strong")

File: PasswordChecker.java
import java.util.Scanner;

public class PasswordChecker {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter your password: ");
        String password = sc.nextLine();

        int length = password.length();
        boolean hasUpper = false;
        boolean hasLower = false;
        boolean hasDigit = false;
        boolean hasSpecial = false;

        String specialCharacters = "!@#$%^&*";

        for (int i = 0; i < password.length(); i++) {
            char c = password.charAt(i);

            if (Character.isUpperCase(c)) {
                hasUpper = true;
            } else if (Character.isLowerCase(c)) {
                hasLower = true;
            } else if (Character.isDigit(c)) {
                hasDigit = true;
            } else if (specialCharacters.indexOf(c) >= 0) {
                hasSpecial = true;
            }
        }

        int score = 0;
        if (length >= 8) score++;
        if (hasUpper) score++;
        if (hasLower) score++;
        if (hasDigit) score++;
        if (hasSpecial) score++;

        System.out.println("\nPassword Analysis");
        System.out.println("Length: " + length);
        System.out.println("Has uppercase letter: " + hasUpper);
        System.out.println("Has lowercase letter: " + hasLower);
        System.out.println("Has number: " + hasDigit);
        System.out.println("Has special character: " + hasSpecial);

        if (score <= 2) {
            System.out.println("\nPassword Strength: Weak");
        } else if (score <= 4) {
            System.out.println("\nPassword Strength: Medium");
        } else {
            System.out.println("\nPassword Strength: Strong");
        }

        sc.close();
    }
}
